moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -83,149 +86,80 @@ planning_pipeline::PlanningPipeline::PlanningPipeline(const moveit::core::RobotM
-    if (!planner_instance_->initialize(robot_model_, node_, parameter_namespace_))
-      throw std::runtime_error("Unable to initialize planning plugin");
-    RCLCPP_INFO(LOGGER, "Using planning interface '%s'", planner_instance_->getDescription().c_str());
-  }
-  catch (pluginlib::PluginlibException& ex)
-  {
-    std::string classes_str = boost::algorithm::join(classes, ", ");
-    RCLCPP_ERROR(LOGGER,
-                 "Exception while loading planner '%s': %s"
-                 "Available plugins: %s",
-                 planner_plugin_name_.c_str(), ex.what(), classes_str.c_str());
-  }
+void planning_pipeline::PlanningPipeline::configure()
+{
+  is_checking_solution_paths_ = false;  // this is set to true below
+  is_publishing_received_requests_ = false;
+  is_displaying_computed_motion_plans_ = false;  // this is set to true below
 
-  // load the planner request adapters
-  if (!adapter_plugin_names_.empty())
-  {
-    std::vector<planning_request_adapter::PlanningRequestAdapterConstPtr> ads;
-    try
-    {
-      adapter_plugin_loader_.reset(new pluginlib::ClassLoader<planning_request_adapter::PlanningRequestAdapter>(
-          "moveit_core", "planning_request_adapter::PlanningRequestAdapter"));
-    }
-    catch (pluginlib::PluginlibException& ex)
-    {
-      RCLCPP_ERROR(LOGGER, "Exception while creating planning plugin loader %s", ex.what());
-    }
+  middleware_handle_->createPlannerPlugin(planner_plugin_name_, robot_model_);
+  middleware_handle_->createAdapterPlugins(adapter_plugin_names_);
 
-    if (adapter_plugin_loader_)
-      for (const std::string& adapter_plugin_name : adapter_plugin_names_)
-      {
-        planning_request_adapter::PlanningRequestAdapterPtr ad;
-        try
-        {
-          ad = adapter_plugin_loader_->createUniqueInstance(adapter_plugin_name);
-        }
-        catch (pluginlib::PluginlibException& ex)
-        {
-          RCLCPP_ERROR(LOGGER, "Exception while loading planning adapter plugin '%s': %s", adapter_plugin_name.c_str(),
-                       ex.what());
-        }
-        if (ad)
-        {
-          ad->initialize(node_, parameter_namespace_);
-          ads.push_back(std::move(ad));
-        }
-      }
-    if (!ads.empty())
-    {
-      adapter_chain_.reset(new planning_request_adapter::PlanningRequestAdapterChain());
-      for (planning_request_adapter::PlanningRequestAdapterConstPtr& ad : ads)
-      {
-        RCLCPP_INFO(LOGGER, "Using planning request adapter '%s'", ad->getDescription().c_str());
-        adapter_chain_->addAdapter(ad);
-      }
-    }
-  }
   displayComputedMotionPlans(true);
   checkSolutionPaths(true);
 }
 
 void planning_pipeline::PlanningPipeline::displayComputedMotionPlans(bool flag)
 {
-  if (display_computed_motion_plans_ && !flag)
-    display_path_publisher_.reset();
-  else if (!display_computed_motion_plans_ && flag)
+  if (is_displaying_computed_motion_plans_ && !flag)
```

Comment:
One of my favorite nitpick topics here :) Please always add braces to explicitly mark the scope. That is, instead of this: 
```
 if(foo)
  bar;
```
add braces so it becomes: 
```
if (foo) {
   bar;
}
```
It helps a lot in eliminating the possibility for bugs stemming from wrong scoping.